test/IRGen/round-trip-debug-types-typealias.swift

```diff
@@ -0,0 +1,10 @@
+// RUN: %target-build-swift -g %s
+
+// https://github.com/apple/swift/issues/66554
+// IRGenDebugInfo type reconstruction crash because existential types
+// inside typealias are not taken into account when comparing type
+// equality
+
+protocol Protocol<T> { associatedtype T }
+typealias AnyProtocol<T> = any Protocol<T>
+let crash: AnyProtocol<Any?>
```

Comment:
Can you pipe the -emit-ir output to FileCheck and CHECK the mangled type name of "crash"

Reply:
@adrian-prantl this is a cherry-pick to 5.9, I think that doing that on main is a good idea though.